### PR TITLE
Nla (mostly client side) cleanup

### PR DESCRIPTION
**Pull request by akallabeth**
* cleans up NLA code for client state machine
* Refactored client side state machine (there were some invalid/broken checks that hid the wrong return from NTLM)
* Add debug logging so it is easy to find at which point the negotiation aborted if something happens

### CODE REVIEW

**Comment by akallabeth** [comment]
Found some memory leaks remaining, postponing

File: libfreerdp/core/nla.c [hunk 1]
```diff
@@ -834,47 +1112,75 @@ static int nla_server_init(rdpNla* nla)
 
 static int nla_server_authenticate(rdpNla* nla)
 {
+	int res = -1;
 	if (nla_server_init(nla) < 1)
-		return -1;
+		goto fail;
+
+	/* Client is starting, here es the state machine:
+	 *
```
Comment (hardening) [inline libfreerdp/core/nla.c:1120]: s/es/is/

File: libfreerdp/core/nla.c [hunk 2]
```diff
@@ -1620,24 +1735,27 @@ static size_t nla_write_ts_password_creds(rdpNla* nla, wStream* s)
...
 	size_t ts_password_creds_length = 0;
-	BOOL ret;
+	BOOL ret = FALSE;
 
-	if (!ts_credentials || !ts_credentials->pvBuffer)
+	if (!data)
 		return FALSE;
 
-	s = Stream_New(ts_credentials->pvBuffer, ts_credentials->cbBuffer);
+	s = Stream_New(data->pvBuffer, data->cbBuffer);
```
Comment (hardening) [inline libfreerdp/core/nla.c:1749]: In this function, we could probably use a static `wStream` for `s` and so avoid the burden of freeing

**Review by hardening — APPROVED** [review]
LGTM

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**